### PR TITLE
Scrollbars

### PR DESCRIPTION
**Pull request by fsih**
### Resolves
Scrollbar part of https://github.com/LLK/scratch-paint/issues/166

### Proposed Changes
Add scrollbars to paint editor

### CODE REVIEW

File: src/components/paint-editor/paint-editor.css [hunk 1]
```diff
@@ -90,10 +90,6 @@ $border-radius: 0.25rem;
     overflow: visible;
 }
 
-.with-eye-dropper {
-    cursor: none;
-}
```
Comment (kchadha) [inline src/components/paint-editor/paint-editor.css:95]: Is this change relevant to this feature?

Comment (kchadha) [reply src/components/paint-editor/paint-editor.css:95]: Ah, never mind, I see it got replaced w/`.hide-cursor` in `scrollable-canvas.css`

File: src/containers/scrollable-canvas.jsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                this.props.canvas.removeEventListener('wheel', this.handleWheel);
+            }
+            nextProps.canvas.addEventListener('wheel', this.handleWheel);
+        }
+    }
+    handleHorizontalScrollbarMouseDown (event) {
+        window.addEventListener('mousemove', this.handleHorizontalScrollbarMouseMove);
+        window.addEventListener('mouseup', this.handleHorizontalScrollbarMouseUp);
+        this.initialMouseX = getEventXY(event).x;
+        this.initialScreenX = paper.view.matrix.tx;
```
Comment (kchadha) [inline src/containers/scrollable-canvas.jsx]: Should this be set before binding the event listeners? I'm wondering if it's possible to have a race condition where `handleHorizontalScrollbarMouseMove` is called before `this.initialMouseX` or `this.initialScreenX` are set...

File: src/containers/scrollable-canvas.jsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        const deltaY = event.deltaY * multiplier;
+        if (event.metaKey || event.ctrlKey) {
+            // Zoom keeping mouse location fixed
+            const canvasRect = this.props.canvas.getBoundingClientRect();
+            const offsetX = event.clientX - canvasRect.left;
+            const offsetY = event.clientY - canvasRect.top;
+            const fixedPoint = paper.view.viewToProject(
+                new paper.Point(offsetX, offsetY)
+            );
+            zoomOnFixedPoint(-deltaY / 1000, fixedPoint);
```
Comment (kchadha) [inline src/containers/scrollable-canvas.jsx:100]: That denominator used to be `100` is the `1000` a typo or intentional?

File: src/containers/scrollable-canvas.jsx (hunk 2 above)
Comment (fsih) [reply src/containers/scrollable-canvas.jsx]: I think it's fine since JS is single-threaded, but I switched them for logical clarity

File: src/containers/scrollable-canvas.jsx (hunk 3 above)
Comment (fsih) [reply src/containers/scrollable-canvas.jsx:100]: Yes, I felt like scroll zoom was way too fast, so I took it down a factor of 10! Sorry, I tried to sneak that in.

File: src/containers/scrollable-canvas.jsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+            this.props.updateViewBounds(paper.view.matrix);
+            this.props.redrawSelectionBox(); // Selection handles need to be resized after zoom
+        } else if (event.shiftKey && event.deltaX === 0) {
+            // Scroll horizontally (based on vertical scroll delta)
+            // This is needed as for some browser/system combinations which do not set deltaX.
+            // See #156.
+            const dx = deltaY / paper.view.zoom;
```
Comment (kchadha) [inline src/containers/scrollable-canvas.jsx:107]: is there a difference between `paper.view.zoom` and `paper.project.view.zoom` ?

Comment (fsih) [reply src/containers/scrollable-canvas.jsx:107]: No, I switched the paper.project.views to paper.views for consistency.

**Review by kchadha — APPROVED** [review]
LGTM!
